lib/unix.bzl

```diff
@@ -0,0 +1,326 @@
+            doc = "Absolute path to the {} command.".format(cmd),
+            mandatory = False,
+        )
+        for cmd in _commands
+    },
+    doc = """
+A toolchain capturing standard Unix shell commands.
+
+Provides:
+  commands: Dict of String, an item per command holding the path to the executable or None.
+  paths: List of String, deduplicated bindir paths. Suitable for generating `$PATH`.
+
+Use `unix_configure` to create an instance of this toolchain.
+See `unix_make_variables` on how to use this toolchain in genrules.
+""",
+    implementation = _unix_toolchain_impl,
+)
+
+def _unix_make_variables_impl(ctx):
+    toolchain = ctx.toolchains[TOOLCHAIN_TYPE]
+    cmd_vars = {
+        "UNIX_%s" % cmd.upper(): cmd_path
+        for cmd in _commands
+        for cmd_path in [toolchain.commands[cmd]]
+        if cmd_path
+    }
+    return [platform_common.TemplateVariableInfo(cmd_vars)]
+
+unix_make_variables = rule(
+    doc = """
+Provides Unix toolchain commands as custom make variables.
+
+Make variables:
+  Provides a make variable of the form `UNIX_<COMMAND>` for each available
+  command, where `<COMMAND>` is the name of the command in upper case.
+
+Use `unix.MAKE_VARIABLES` instead of instantiating this rule yourself.
+
+Example:
+  >>> genrule(
+          name = "use-grep",
+          srcs = [":some-file"],
+          outs = ["grep-out"],
+          cmd = "$(UNIX_GREP) search $(execpath :some-file) > $(OUTS)",
+          toolchains = [unix.MAKE_VARIABLES],
+      )
+""",
+    implementation = _unix_make_variables_impl,
+    toolchains = [TOOLCHAIN_TYPE],
+)
+
+def _unix_config_impl(repository_ctx):
+    cpu = get_cpu_value(repository_ctx)
+    commands = {}
+    for cmd in _commands:
+        cmd_path = repository_ctx.which(cmd)
+        if cmd_path == None and cpu == "x64_windows":
+            cmd_path = repository_ctx.which(cmd + ".exe")
+        commands[cmd] = cmd_path
+    repository_ctx.file("BUILD.bazel", executable = False, content = """
+load("@bazel_skylib//lib:unix.bzl", "unix_toolchain")
+unix_toolchain(
+    name = "local_unix",
+    visibility = ["//visibility:public"],
+    {commands}
+)
+toolchain(
+    name = "local_unix_toolchain",
+    toolchain = ":local_unix",
+    toolchain_type = "{toolchain_type}",
+    exec_compatible_with = [
+        "@bazel_tools//platforms:x86_64",
+        "@bazel_tools//platforms:{os}",
+    ],
+    target_compatible_with = [
+        "@bazel_tools//platforms:x86_64",
+        "@bazel_tools//platforms:{os}",
+    ],
+)
+""".format(
+        commands = ",\n    ".join([
+            '{cmd} = "{path}"'.format(cmd = cmd, path = cmd_path)
+            for (cmd, cmd_path) in commands.items()
+            if cmd_path
+        ]),
+        os = {
+            "darwin": "osx",
+            "x64_windows": "windows",
+        }.get(cpu, "linux"),
+        toolchain_type = TOOLCHAIN_TYPE,
+    ))
+
+_unix_config = repository_rule(
+    environ = ["PATH"],
+    local = True,
```

Comment:
This [repository rule](https://docs.bazel.build/versions/master/skylark/lib/globals.html#repository_rule) is a configure rule. Please add `configure=True`, so that it gets also invalidated upon `bazel sync --configure`.